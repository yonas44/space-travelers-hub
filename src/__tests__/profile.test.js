import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import ProfilePage from '../components/Profile';

describe('Profile loads', () => {
  test('the component renders', () => {
    render(
      <Provider store={store}>
        <ProfilePage />
      </Provider>,
    );
    expect(screen.getByText(/My Missions/)).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const renderComponents = renderer
      .create(
        <Provider store={store}>
          <ProfilePage />
        </Provider>,
      )
      .toJSON();
    expect(renderComponents).toMatchSnapshot();
  });
});
