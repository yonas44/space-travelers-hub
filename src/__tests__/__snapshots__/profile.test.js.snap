// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile loads matches snapshot 1`] = `
<div
  className="profile-wrapper"
>
  <div
    className="mission-section"
  >
    <h3
      className="rockets"
    >
      My Missions
    </h3>
    <table
      className="mission-table"
    >
      <tbody>
        <tr>
          <td
            className="empty-mission"
          >
            You have not joined any mission yet!
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <div
    className="rockets-section"
  >
    <h3>
      My Rockets
    </h3>
    <table
      className="rocket-table"
    >
      <tbody>
        <tr>
          <td
            className="no-rockets"
          >
            No reservations
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
